src/Broadway/EventSourcing/AggregateFactory/AggregateFactoryInterface.php

```diff
@@ -15,5 +16,5 @@
      *
      * @return \Broadway\EventSourcing\EventSourcedAggregateRoot
      */
-    public function create($aggregateClass, DomainEventStreamInterface $domainEventStream);
+    public function create($aggregateClass, DomainEventStreamInterface $domainEventStream, DomainMessage $snapshot);
```

Comment:
I suggest to use concrete `Snapshot` class for this purpose, as `DomainMessage` could lead to misuse.